### PR TITLE
Resolve the device path before generating partition layout

### PR DESCRIPTION
**Pull request by johnliu55tw**
Without resolving it first, elemental installation would fail.

## Test plan
1. Install Harvester with any node using PXE boot installation with setting `install.device` to /dev/disk/by-path instead of `/dev/vda`
2. Make sure installation succeeded

## Related issues
- https://github.com/harvester/harvester/issues/2256

### CODE REVIEW

File: pkg/config/cos.go [hunk 1]
```diff
@@ -537,6 +537,11 @@ func CreateRootPartitioningLayout(devPath string) (*yipSchema.YipConfig, error)
 		return nil, err
 	}
 
+	resolvedDevPath, err := filepath.EvalSymlinks(devPath)
```
Comment (bk201) [inline pkg/config/cos.go:540]: Does `EvalSymlinks` also work for normal device files?

Comment (johnliu55tw) [reply pkg/config/cos.go:540]: Yes. If it's not a symlink it would just return the original value.

**Review by bk201 — APPROVED** [review]
LGTM

**Review by weihanglo — APPROVED** [review]
Looks reasonable. Thanks!

**Pull request merged**